lib/curl_ntlm_core.h

```diff
@@ -41,6 +41,11 @@
 #  include <openssl/ssl.h>
 #endif
 
+// enable USE_NTLM2SESSION even if USE_WIN32_CRYPTO is defined.
+// for NTML auth for SMTP, we need USE_NTLM2SESSION
```

Comment:
Typo: NTML -> NTLM.
And why not just remove the `&& !defined(USE_WIN32_CRYPTO)` below?

Reply:
Well, I am not sure if removing the USE_WIN32_CRYPTO check would do the same.
USE_NTRESPONSES may not be defined.

Feel free to fix the typo or adjust the #if lines.

Reply:
I'm with @MarcelRaad on this. `USE_NTLM2SESSION` seems to be used in the code only if `USE_NTRESPONSES` also is set (in vauth/ntlm.c) so it seems to rely on that fact, which then makes it a simpler change.

Reply:
For me USE_WIN32_CRYPTO and USE_OPENSSL are defined.
OPENSSL_NO_MD4 is not defined, so USE_NTRESPONSES is set here.

removing    && !defined(USE_WIN32_CRYPTO)   seems also to do it.

Reply:
Closing as I made another pull request.